### PR TITLE
[BugFix] Fix cancel rewrite acutally error message

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/StarRocksTest/issues/1980

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by meegoo** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java [hunk 1]
```diff
@@ -1339,7 +1339,9 @@ public void cancel() {
     }
 
     private void cancelInternal(PPlanFragmentCancelReason cancelReason) {
-        connectContext.getState().setError(cancelReason.toString());
+        if (connectContext.getState().getErrorMessage().equals("")) {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java:1342]: empty()?

**Review by Astralidea — APPROVED** [review]

**Comment by meegoo** [comment]
run all

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java [hunk 2]
```diff
@@ -1339,7 +1339,9 @@ public void cancel() {
     }
 
     private void cancelInternal(PPlanFragmentCancelReason cancelReason) {
-        connectContext.getState().setError(cancelReason.toString());
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java:1342]: If the errMsg is null, it will report NullPointerException.

**Review by gengjun-git — APPROVED** [review]

**Comment by meegoo** [comment]
run starrocks_fe_unittest

**Comment by meegoo** [comment]
run starrocks_fe_unittest

**Comment by meegoo** [comment]
run starrocks_fe_unittest

**Comment by meegoo** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|2|2|100.00%|[]|

**Comment by meegoo** [comment]
@merfigy backport branch-3.0

**Comment by meegoo** [comment]
@mergify backport branch-2.4

**Pull request merged**